pkg/processor/sumologicschemaprocessor/processor.go

```diff
@@ -64,42 +84,36 @@ func (processor *sumologicSchemaProcessor) shutdown(_ context.Context) error {
 }
 
 func (processor *sumologicSchemaProcessor) processLogs(_ context.Context, logs plog.Logs) (plog.Logs, error) {
-	logs, err := processor.cloudNamespaceProcessor.processLogs(logs)
-	if err != nil {
-		return logs, err
-	}
+	for i := 0; i < len(processor.subprocessors); i++ {
+		err := processor.subprocessors[i].processLogs(logs)
 
-	logs, err = processor.translateAttributesProcessor.processLogs(logs)
-	if err != nil {
-		return logs, err
+		if err != nil {
+			return logs, err
+		}
```

Comment:
It can be shortened:

```go
if err := processor.subprocessors[i].processLogs(logs); err != nil {
    return logs, err
}
```

Reply:
I guess the refactor now makes it more difficult to track which one subprocessor returned the error. I thought that the errors would contain that info (as for now, there's nothing that can go wrong inside them).

Reply:
> I guess the refactor now makes it more difficult to track which one subprocessor returned the error

Let's use the name we just added for error propagation

```
fmt.Errorf("failed to process by %s: %v", processor.ConfigPropertyName(), err)
```